### ISSUE TITLE
Checked and unchecked tasks still cause boss quest damage and task-based attribute training

### ISSUE THREAD

**Issue by Pexpy** [HabitRPG/habitica#5648]
Just had a day where I completed a large to-do - 4 checklist items, task value -150.96. Over the course of the day I checked and unchecked it three times. Once as Medium difficulty, another as Easy, once as Medium with a 106% crit and lastly the final completed value as a Medium to-do with no crit.

After checking it three times - Medium, Easy, Medium Crit - I had an estimated 60 damage on the boss on Data Display. This stayed after I unchecked it.
Checking it the fourth and final time, the damage went up to around 85.

I had figured this would fix itself in cron and deal what should be around my general daily damage of 9 + the 25 damage from that one task but the next day I did 83 damage which implies it's stacking up each time.

I might have a look at testing this in a closed environment but I'll need to get some gems first for a boss with enough health.

User ID: 4352e75b-0c23-4aa0-b64c-ea82543aa2a3
Group ID: 3918426d-5a0a-42bf-8247-737e890fe32d

**Note** added June 2018: This issue also affect attribute training points for task-based automatic stat allocation.

**Comment by Alys** [comment]
Yes, this is a known problem. The code that collates boss damage from tasks doesn't subtract it when the task is unticked. I'm not sure we have an issue for it yet so I'll keep this one open. We've always considered it very low priority since it usually has little effect on total boss damage.

**Comment by librarianmage** [comment]
@Alys, is this fixed? It looks like @doyler added a comment that fixed it.

**Comment by Alys** [comment]
@MathWhiz That's a commit on doyler's fork of github, and we can't review it until there's been a pull request (that triggers automatic tests that help us assess the code).

@doyler We're currently making changes to the API, which includes the code you've modified in your commit. The changes will be finished reasonably soon now, so if you want to continue with this fix, you might like to wait until the new API is live. If you're a member of the Aspiring Coders guild, you'll see a notification there when it's live.

**Comment by librarianmage** [comment]
@Alys Oh! I missed that.

**Comment by doyler** [comment]
Yea I submitted the pull request but it was sitting around for awhile since there were no tests for it.

Gotcha though, I guess I'll wait for the new API then, but this has already been fixed on my end!

Thanks.

**Comment by Alys** [comment]
For the record, this is the PR (currently closed pending tests, as doyler says): https://github.com/HabitRPG/habitrpg/pull/6502

**Comment by Alys** [comment]
I'm most of the way through refactoring common/script/ops/scoreTask.js which contains the code for this, so I'm assigning this to myself.

**Comment by natfarleydev** [comment]
Does this need (re)doing at all?

**Comment by veeeeeee** [comment]
Given that #6502 has been closed and merged, does this issue still need to remain open?

**Comment by Alys** [comment]
That PR was closed without being merged - the text at the bottom about the branch being successfully deployed is misleading. There ended up being no changes in that PR due to the force-push of commit that reversed them.

**Comment by michkrej** [comment]
Hi! I could take a look at this if it still needs to be fixed :) Otherwise I would appreciate if you could point me in the direction of another ticket, I would love to help out!
